src/webgl/p5.RendererGL.js

```diff
@@ -83,6 +81,93 @@ p5.RendererGL.prototype._initContext = function() {
     throw new Error(er);
   }
 };
+
+//This is helper function to reset the context anytime the attributes
+//are changed with setAttributes()
+
+p5.RendererGL.prototype._resetContext = function(attr, options, callback) {
+  var w = this.width;
+  var h = this.height;
+  var defaultId = 'defaultCanvas0';
+  var c = document.getElementById(defaultId);
+  if(c){
+    c.parentNode.removeChild(c);
+  }
+  c = document.createElement('canvas');
+  c.id = defaultId;
+  if (this._pInst._userNode) {
+    this._pInst._userNode.appendChild(c);
+  } else {
+    document.body.appendChild(c);
+  }
+  this._pInst.canvas = c;
+  this._pInst._setProperty('_renderer', new p5.RendererGL(this._pInst.canvas,
+    this._pInst, true, attr));
+  this._pInst._isdefaultGraphics = true;
+  this._pInst._renderer.resize(w, h);
+  this._pInst._renderer._applyDefaults();
+  this._pInst._elements.push(this._renderer);
+  if(typeof callback === 'function') {
+    //setTimeout with 0 forces the task to the back of the queue, this ensures that
+    //we finish switching out the renderer
+    setTimeout(function() {
+      callback.apply(window._renderer, options);
+    }, 0);
+  }
+};
+
+
+/**
+ *
+ * Set attributes for the WebGL Drawing context.
+ * This should be put in setup()
+ * The available attributes are
+ * alpha - indicates if the canvas contains an alpha buffer
+ * default is true
+ *
+ * depth - indicates whether the drawing buffer has a depth buffer
+ * of at least 16 bits - default is true
+ *
+ * stencil - indicates whether the drawing buffer has a stencil buffer
+ * of at least 8 bits
+ *
+ * antialias - indicates whether or not to perform anti-aliasing
+ * default is false
+ *
+ * premultipliedAlpha - indicates that the page compositor will assume
+ * the drawing buffer contains colors with pre-multiplied alpha
+ * default is false
+ *
+ * preserveDrawingBuffer - if true the buffers will not be cleared and
+ * and will preserve their values until cleared or overwritten by author
+ * (note that p5 clears automatically on draw loop)
+ * default is true
+ *
+ * @method setAttributes
+ * @param  {String|Object}  String name of attribute or object with key-value pairs
+ * @param  {Boolean}        New value of named attribute
+ *
```

Comment:
can we add an example or two? you can use <div class='norender'> if the visual result doesn't display anything useful. i've found it can be helpful for people just to see the code written in context. we could add a line of comment after the call in the example explaining what difference should be noticeable in a running sketch.